src/datasource.ts

```diff
@@ -396,6 +404,31 @@ export class OpenSearchDatasource extends DataSourceWithBackend<OpenSearchQuery,
     return [query, ...osFilters].filter(f => f).join(' AND ');
   }
 
+  addPPLAdHocFilters(queryString: string, adHocFilters: any) {
+    for (let i = 0; i < adHocFilters.length; i++) {
+      const { key, operator } = adHocFilters[i];
+      let { value } = adHocFilters[i];
+
+      if ('=~' === operator || '!~' === operator || !key || !value) {
```

Comment:
Would you explain more about this condition? What's that about the operators? Was this already illustrated in a test case and I missed it?

Reply:
this wasn't in a test case before. these two operators are not recognized by opensearch so it returns an error if you try to use them.

Reply:
are there tests now documenting this? if not, would it be worth it to add them? honestly asking that and deferring to your judgment about this!

Reply:
https://github.com/grafana/opensearch-datasource/pull/244/files#diff-f5855f08029d58a7871f69a0ebfe07376b2508725c81159df890fc793eb5b545R1629-R1630 there's this test case